sxg_rs/src/link.rs

```diff
@@ -14,6 +15,94 @@ pub(crate) async fn process_link_header(
     fallback_url: &Url,
     header_integrity_fetcher: &mut dyn HeaderIntegrityFetcher,
 ) -> String {
+    let links = match parse_link_header(value) {
+        Ok(links) => links,
+        Err(_) => {
+            return "".into();
+        }
+    };
+
+    let (preloads, allowed_alt_sxgs) = preloads_and_allowed_alt_sxgs(links, fallback_url);
+
+    let fallback_origin = fallback_url.origin();
+    let directives = RefCell::new(vec![]);
+    stream::iter(preloads)
+        .for_each_concurrent(None, |link| async {
+            let link = link;
+            let srcset = get_param(&link.params, "imagesrcset");
+            let images = match &srcset {
+                Some(srcset) => srcset::parse(srcset).unwrap_or_default(),
+                None => vec![],
+            };
+            // Convert image hrefs from srcset to absolute, for comparison with
+            // allowed_alt_sxgs.
+            let images = images
+                .iter()
+                .filter_map(|url| fallback_url.join(url).ok().map(String::from));
```

Comment:
Does this work when the URL in `srcset` is already an absolute URL?

Reply:
Yes: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8dc6a1bd8b4c0ecbea3a597259f9f0e0